packages/opentelemetry-resources/src/detectors/BrowserDetector.ts

```diff
@@ -61,5 +62,18 @@ class BrowserDetector implements Detector {
   }
 }
 
+//Add Browser related attributes to resources
+function getBrowserAttributes(): ResourceAttributes {
+  const browserAttribs : ResourceAttributes = {};
+  const userAgentData : UserAgentData | undefined = (navigator as any).userAgentData;
+  if(userAgentData) {
+    browserAttribs[BROWSER_ATTRIBUTES.PLATFORM] = userAgentData.platform;
+    browserAttribs[BROWSER_ATTRIBUTES.BRANDS] = userAgentData.brands;
+    browserAttribs[BROWSER_ATTRIBUTES.MOBILE] = userAgentData.mobile;
+  }
+  browserAttribs[BROWSER_ATTRIBUTES.USER_AGENT] = navigator.userAgent;
```

Comment:
@martinkuba do we need to include the `user_agent` attribute even when `platform`, `brands` and `mobile` are present or only when they are not available?

Reply:
The intent was to include it only when the userAgentData API is not available, so that backends move to using the new API. Parsing the userAgent string should only be done to support old browsers.

Reply:
Should I then remove the user_agent attribute. The runtime version attribute is always added and it is the userAgent, so browsers that don't support userAgentData can parse it.

Reply:
The `process.runtime.*` attributes should be deprecated in browser resource attributes now that we have `browser.*` attributes in the spec. However, `browser.user_agent` should be included only in an else block when `userAgentData` isn't supported by the browser.

I wonder what the process for this deprecation is. Just add to changelog?

Reply:
The `process.runtime.version` attribute probably should no longer be used for browsers, so that the user-agent string is not sent twice. But I am not sure that `process.runtime.name` should be deprecated. All of the `browser` attributes are optional, so it is possible that they would not be sent if the browser does not provide them. On the other hand, the value for `process.runtime.name` is the static value `browser`; it is added by the SDK without detecting anything, which makes it more reliable.

Reply:
I checked with @cuichenli offline and got to know that the `process.runtime.*` attributes are coming from the spec [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md#javascript-runtimes).   Further, it doesn't look like anyone is using it (I checked with @pkanal, @t2t2 and @cuichenli ), so it is safe to remove them if we agree on it and the spec for `process.runtime` semconv is updated removing browser related text.

We discussed these in the 9/7 RUM SIG today - 

> All of the browser attributes are optional, so it is possible that they would not be sent if the browser does not provide them

This is not possible as `navigator.userAgent` is [well supported](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent#browser_compatibility) on all browsers although its value is going to have less information going forward. And as per the change in this PR, we are also going to rely on `navigator.userAgentData` which, although is experimental, will continue to be available in browsers going forward. so between `userAgent` and `userAgentData`,  at least one of them will always be available as they are part of the standard browser API specs.

> On the other hand, the value for process.runtime.name is the static value browser; it is added by the SDK without detecting anything, which makes it more reliable.

There are alternate ways of identifying the telemetry as coming from browsers - 
- `telemetry.sdk.language` having value `webjs`
- presence of an attribute with namespace `browser`


@legendecas since you introduced the browser language in the spec [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/process.md#javascript-runtimes), can you weigh in on removing these attributes from BrowserDetector? They are not required given we are introducing browser.* attributes, and no one seems to be using it already.  If you agree, I will submit a PR removing the browser mention from the javascript-runtimes.

Reply:
@legendecas Can you respond to my comment above? We could discuss it in the JS SIG too this week.

Reply:
Sorry for the late response.

The intention of the specified attributes is to write down what's already shipped.

As you mentioned removing existing attributes can be breaking. Package @opentelemetry/resource is declared stable so I'm not sure about introducing such a change to the existing API is fine. One possible solution is to add a new detector like `UserAgentDataDetector`, which implements the newer `browser.*` attributes, and in the meantime, we can mark the `BrowserDetector` as deprecated.

Reply:
Yes, if we need to avoid reporting the legacy process attributes for the browser, we may add a new detector to not introduce breakage.

Reply:
We discussed in the JS SIG today 9/14 - it was suggested that we add the new detector in a separate package in the experimental folder so that its release can be managed independently. 

Separately, I will submit a PR to the spec repo and get the process.runtime.* attributes removed for the browser case, after we which we can mark the `BrowserDetector` in this package as deprecated.